integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -363,50 +350,47 @@ def _normalize_filters(filters: Dict[str, Any]) -> Tuple[List[str], Dict[str, An
         passed to `ids`, `where` and `where_document` respectively.
         """
         if not isinstance(filters, dict):
-            msg = "'filters' parameter must be a dictionary"
+            msg = f"'{filters}' parameter must be a dictionary"
             raise ChromaDocumentStoreFilterError(msg)
 
         ids = []
         where = defaultdict(list)
-        where_document = defaultdict(list)
-        keys_to_remove = []
+        where_document = {}
+        document_flag = False
 
-        for field, value in filters.items():
+        # Process filters
+        for field, value in list(filters.items()):
             if field == "content":
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
                 where_document["$contains"] = value
+                document_flag = True
+                del filters[field]
             elif field == "id":
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
                 ids.append(value)
-            elif isinstance(value, (list, tuple)):
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
-
-                # if the list is empty the filter is invalid, let's just remove it
-                if len(value) == 0:
+                del filters[field]
+            elif isinstance(value, (list, tuple)) and field not in ["$and", "$or"]:
+                if not value:  # Skip empty lists
+                    del filters[field]
                     continue
 
-                # if the list has a single item, just make it a regular key:value filter pair
                 if len(value) == 1:
-                    where[field] = value[0]
-                    continue
-
-                # if the list contains multiple items, we need an $or chain
-                for v in value:
-                    where["$or"].append({field: v})
-
-        for k in keys_to_remove:
-            del filters[k]
-
-        final_where = dict(filters)
-        final_where.update(dict(where))
+                    where[field] = value[0]  # Simplify single-item lists
+                # Check if list contains special operators that apply to documents
+                elif any(isinstance(v, dict) and any(k in v for k in ["$contains", "$not_contains"]) for v in value):
+                    where_document = filters.copy()
```

Comment:
Why are we copying the partially modified filters dict here? This will lead to different outputs depending on the ordering of the dict items.